### DISCUSSION TITLE
Relative/absolute imports policy (e.g. in `AnalysisBase`)?

### DISCUSSION THREAD

**Discussion by marinegor** [MDAnalysis/mdanalysis#4434] [Development]
Hi everyone, I noticed that some core code uses relative imports (like `core/universe.py`), and some -- absolute (like `analysis/base.py`):

```python
# core/universe.py
from .. import _TOPOLOGY_ATTRS, _PARSERS
from ..exceptions import NoDataError
from ..lib import util
from ..lib.log import ProgressBar
from ..lib.util import cached, NamedStream, isstream
from ..lib.mdamath import find_fragments
from . import groups
from ._get_readers import get_reader_for, get_parser_for
from .groups import (ComponentBase, GroupBase,
                     Atom, Residue, Segment,
                     AtomGroup, ResidueGroup, SegmentGroup)

# analysis/base.py
import numpy as np
from MDAnalysis import coordinates
from MDAnalysis.core.groups import AtomGroup
from MDAnalysis.lib.log import ProgressBar
```

On my system (vscode + pylance), the latter results in "module not found" warnings and poorer autocomplete and typechecking. Is there a particular reason on why it should be relative that I don't know, like some CI/CD things?

**Comment by hmacdope** [comment]
I think its just a hangover from different people's styles. I don't mind either way really.

**Comment by orbeckst** [comment]
I think relative imports are generally better in a package; in principle the `import MDAnalysis` could be a different package in the system.

However, the `MDAnalysis.analysis` module is effectively separate from the rest of the MDA core so it makes some sense to pretend that it is importing stuff from "outside". 

In the absence of rules I'd 
- use relative important _inside analysis_ and _inside MDAnalysis without analysis_
- use absolute imports when code _inside analysis_ imports from MDAnalysis core modules

(Btw, when we say "MDAnalysis core" we mean more than just "MDAnalysis.core", it's really "everything except analysis.)
